include/bm/bm_sim/psa_counter.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2013-present Barefoot Networks, Inc.
```

Comment:
This is the psa equivalent of a counter_array, extends the extern type. P_Counter is literally identical to the Counter defined in counter.h. I believe @jnfoster said at the end of the day, we would like to not intertwine the implementations of simple switch counters and psa counters (I may have also misunderstood what he said).